File: server/java/src/main/java/com/onlinepayments/example/service/domain/paymentdetails/AddressPersonal.java
package com.onlinepayments.example.service.domain.paymentdetails;

public record AddressPersonal(
        String additionalInfo,
        String city,
        String companyName,
        String countryCode,
        String houseNumber,
        PersonalName name,
        String state,
        String street,
        String zip
) { }

File: server/nodejs/src/business/domain/payments/payment-details/address-personal.ts
import { PersonalName } from './personal-name';

export type AddressPersonal = {
    additionalInfo?: string | null;
    city?: string | null;
    companyName?: string | null;
    countryCode?: string | null;
    houseNumber?: string | null;
    name?: PersonalName | null;
    state?: string | null;
    street?: string | null;
    zip?: string | null;
};
